vignettes/VectorControl_IRS.Rmd

```diff
@@ -0,0 +1,158 @@
+---
+title: "Vector Control: Indoor Residual Spraying"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Vector Control: IRS}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(malariasimulation)
+library(malariaEquilibrium)
+```
+
+Indoor Residual Spraying (IRS) involves periodically treating indoor walls with insecticides to eliminate adult female mosquitoes that rest indoors. `malariasimulation` can be used to investigate the effect of malaria control strategies that deploy IRS. The model has the functionality to allow users to specify a range of IRS campaigns, including different timings and coverages and in settings with specified proportions of mosquito species with the help of the `set_spraying()` function. 
+
+### Plotting functions
+We will create a few plotting functions to visualise the output.
+```{r}
+cols <- c("#E69F00", "#56B4E9", "#009E73", 
+          "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
+
+# Plotting functions
+plot_prev <- function() {
+  plot(x = output$timestep, y = output$n_detect_730_3650/output$n_730_3650, 
+       type = 'l', col = cols[3], lwd = 2.5,
+       xlab = 'Time (days)', ylab = expression(paste(italic(Pf),"PR"[2-10])),
+       xaxs = "i", yaxs = "i", bty = "l", ylim = c(0,1))
+  lines(x = output_control$timestep, y = output_control$n_detect_730_3650/output_control$n_730_3650,
+        col = cols[5], lwd = 2.5)
+  abline(v = sprayingtimesteps, lty = 2, lwd = 2.5, col = 'black')
+  grid(lty = 2, col = 'grey80', lwd = 1)
+  legend('bottomleft', box.lty = 0, legend = c('Spraying', 'Prevalence for IRS scenario','Prevalence for control scenario'),
+         col = c('black', cols[3], cols[5]), lty = c(2,1,1), lwd = 2.5)
+}
+```
+
+### Parameterisation 
+
+Use the `get_parameters()` function to generate the list of parameters for a seasonal profile, accepting the default values to run the simulation from an equilibrium starting point. The seasonality will drive mosquito dynamics in the absence of interventions. 
+
+```{r}
+year <- 365
+month <- 30
+sim_length <- 3 * year
+human_population <- 1000
+starting_EIR <- 50
+
+simparams <- get_parameters(
+  list(
+    human_population = human_population,
+    # seasonality parameters
+    model_seasonality = TRUE, # Let's try a bi-modal model
```

Comment:
Unlike with bed nets, seasonality is a defining factor of the IRS campaign, so it should be left in here 👍

Reply:
To make it easier further on to work with the timing relative to the peak, we should probably use seasonality parmeters for a simple seasonal site, not bi-modal as the comment suggests.